### PR TITLE
Convert home sample cards to react component

### PR DESCRIPTION
**Pull request by NMilivojevic**
I have created `WishCardsCarousel.jsx` component in `home` folder. Bootstrap classes from `homeSampleCards.pug` are used there. 

A separate `WishCard.jsx` component is made in `shared` folder and it returns only the card with image and text below it. 
This component is used in `wishcards/index.jsx` and in `home/WishCardsCarousel.jsx`. It receives as props the full `wishCard` object.

In place of `homeSampleModal.pug` in `home.pug` page, is a `#carousel-root,` which is used below in js to mount `WishCardsCarousel.jsx`. The wishCards response from backend is passed to it. Should the user object response be passed to it as well?

### CODE REVIEW

**Comment by Enubia** [comment]
> Should the user object response be passed to it as well?

I don't think it's needed

**Review by Enubia — APPROVED** [review]
LGTM! @stacysealky please check it as well

**Comment by NMilivojevic** [comment]
> > Should the user object response be passed to it as well?
> 
> I don't think it's needed

There was one instance of conditional rendering based if logged in user clicked on `Donate Gift` button, or public user did. In `homeSampleCards.pug` it went like this
```
if user
a( href='wishcards/donate/' + currentCard._id )
button.....Donate Gift
else
button.... Donate Gift
                                                        
```

Not sure if that conditional rendering is still relevant.

File: js/react/utils/helpers.jsx [hunk 1]
```diff
@@ -6,3 +6,11 @@ export const loadlGoogleClientLibraryScript = (src) => {
...
 	body.appendChild(tag);
 };
+
+export const chunkArray = (arr, chunkSize) => {
+	const chunkedArray = [];
+	for (let i = 0; i < arr.length; i += chunkSize) {
+		chunkedArray.push(arr.slice(i, i + chunkSize));
+	}
+	return chunkedArray;
+};
```
Comment (nick-w-nick) [inline js/react/utils/helpers.jsx:16]:
```suggestion
export const chunkArray = (initialArray, chunkSize) => {
	const chunks = [];
	for (let index = 0; index < initialArray.length; index += chunkSize) {
		const chunk = initialArray.slice(index, index + chunkSize);
		chunks.push(chunk);
	}
	
	return chunks;
};
```

Minor readability improvements

**Comment by Enubia** [comment]
> > > Should the user object response be passed to it as well?
> 
> > 
> 
> > I don't think it's needed
> 
> 
> 
> There was one instance of conditional rendering based if logged in user clicked on `Donate Gift` button, or public user did. In `homeSampleCards.pug` it went like this
> 
> ```
> 
> if user
> 
> a( href='wishcards/donate/' + currentCard._id )
> 
> button.....Donate Gift
> 
> else
> 
> button.... Donate Gift
> 
>                                                         
> 
> ```
> 
> 
> 
> Not sure if that is still relevant.

I think for now it's not needed, if we find out during testing that it should be added again we can do that in a new task

**Comment by stacysealky** [comment]
@NMilivojevic Looks great!! Will merge after the above comments are addressed. Thanks!

**Comment by Enubia** [comment]
resolves #558

**Pull request merged**